From: Noa Osherovich <noaos@mellanox.com>
Date: Thu, 20 Apr 2017 20:53:31 +0300
Subject: IB/core: Add HDR speed enum
Patch-mainline: v4.12-rc1
Git-commit: 12113a35ada6bba074836d3d26671213e12069bf
References: bsc#1015342 FATE#321688 bsc#1015343 FATE#321689

Add high data rate speed to the ib_port_speed enumeration.

Signed-off-by: Noa Osherovich <noaos@mellanox.com>
Signed-off-by: Eran Ben Elisha <eranbe@mellanox.com>
Signed-off-by: Leon Romanovsky <leon@kernel.org>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/infiniband/core/sysfs.c |    4 ++++
 include/rdma/ib_verbs.h         |    3 ++-
 2 files changed, 6 insertions(+), 1 deletion(-)

--- a/drivers/infiniband/core/sysfs.c
+++ b/drivers/infiniband/core/sysfs.c
@@ -252,6 +252,10 @@ static ssize_t rate_show(struct ib_port
 		speed = " EDR";
 		rate = 250;
 		break;
+	case IB_SPEED_HDR:
+		speed = " HDR";
+		rate = 500;
+		break;
 	case IB_SPEED_SDR:
 	default:		/* default to SDR for invalid rates */
 		rate = 25;
--- a/include/rdma/ib_verbs.h
+++ b/include/rdma/ib_verbs.h
@@ -416,7 +416,8 @@ enum ib_port_speed {
 	IB_SPEED_QDR	= 4,
 	IB_SPEED_FDR10	= 8,
 	IB_SPEED_FDR	= 16,
-	IB_SPEED_EDR	= 32
+	IB_SPEED_EDR	= 32,
+	IB_SPEED_HDR	= 64
 };
 
 /**
